conformance/lang/expressions/list_constructor.balt

```diff
@@ -0,0 +1,1311 @@
+}
+
+Test-Case: error
+Description: Test the compile-time error when the applicable contextually expected type does not leave
+             a single list type descriptor.
+Fail-Issue: ballerina-platform/ballerina-lang#33056
+Labels: list-constructor-expr, array-type, tuple-type, union-type, byte, string, boolean
+
+function errorFunction() {
+    string[]|int[]|boolean[] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+    string[]|int[] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+
+    [int, int, int]|[string...] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+    [int, int, int]|[string...]|int|boolean _ = [1, 2, 3]; // @error inherent type cannot be a union type
+
+    [int, int, int]|boolean[] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+    int[]|[boolean] _ = [1, 2, 3]; // @error inherent type cannot be a union type
+}
+
+Test-Case: output
+Description: Test when there is no contextually expected type, then the inherent type is a a tuple-type-descriptor
+             with a member-type-descriptor for each expression in the expr-list,
+             where the type of each member-type-descriptor will be the broad type of the corresponding expression
+             in the expr-list.
+Labels: list-constructor-expr, is-expr, var, boolean-literal, singleton-type, DecimalNumber, tuple-type, string
+
+function init() {
+    var a1 = [1, 2, 3];
+    io:println(a1 is [int, int, int]); // @output true
+    io:println(a1 is [byte, byte, byte]); // @output false
+
+    var a2 = [1, "foo", "bar", 3];
+    io:println(a2 is [int, string, string, int]); // @output true
+    io:println(a2 is [byte, "foo", "bar", byte]); // @output false
+
+    var a3 = [1, ["foo", [true, 23]], [true, false, true]];
+    io:println(a3 is [int, [string, [boolean, int]], [boolean, boolean, boolean]]); // @output true
+    io:println(a3 is [1, ["foo", [true, 23]], [true, false, true]]); // @output false
+
+    io:println([1, 2, 3] is [int, int, int]); // @output true
+    io:println([1, 2, 3] is [1, 2, 3]); // @output false
+
+    io:println([1, "foo", "bar", 3] is [int, string, string, int]); // @output true
+    io:println([1, "foo", "bar", 3] is [1, "foo", "bar", 3]); // @output false
+
+    io:println([1, ["foo", [true, 23]], [true, false, true]] is [int, [string, [boolean, int]], [boolean, boolean, boolean]]); // @output true
+    io:println([1, ["foo", [true, 23]], [true, false, true]] is [1, ["foo", [true, 23]], [true, false, true]]); // @output false
+}
+
+Test-Case: error
+Description: Test when there is a contextually expected type, then the type that the inherent type
+             requires for each list member provides the contextually expected type for the expression for the member.
+Labels: list-constructor-expr, member-access-expr, array-type, DecimalFloatingPointNumber, byte, float, union-type,
+        optional-type, unary-minus
+
+function errorFunction() {
+    byte[] a1 = [1, 2, 3];
```

Comment:
Contextually-expected type is more of a compile time concept. IMO, the tests for this sentence should be those having invalid member expressions in the list constructor.

```ballerina
function errorFunction() {
    int[] _ = [
        "", // @error expected 'int', found 'string'
        1.0 // @error expected 'int', found 'float'
    ];
}
```